statsmodels/miscmodels/tests/test_ordinal_model.py

```diff
@@ -0,0 +1,367 @@
+        assert_allclose(resf2.params, resp.params, atol=1e-8)
+        assert modf2.exog_names == resp.model.exog_names
+        assert modf2.data.ynames == resp.model.data.ynames
+        assert hasattr(modf2.data, "frame")
+        assert not hasattr(modf2, "frame")
+
+        with pytest.raises(ValueError):
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore", SpecificationWarning)
+                OrderedModel.from_formula(
+                    "apply ~ pared + public + gpa - 1",
+                    data={"apply": np.asarray(data['apply']),
+                          "pared": data['pared'],
+                          "public": data['public'],
+                          "gpa": data['gpa']},
+                    distr='probit')
+
+
+class TestLogitModelFormula():
+
+    @classmethod
+    def setup_class(cls):
+        warnings.simplefilter("ignore", SpecificationWarning)
+        data = ds.df
+        nobs = len(data)
+        data["dummy"] = (np.arange(nobs) < (nobs / 2)).astype(float)
+        # alias to correspond to patsy name
+        data["C(dummy)[T.1.0]"] = data["dummy"]
+        cls.data = data
+
+        columns = ['C(dummy)[T.1.0]', 'pared', 'public', 'gpa']
+        # standard fit
+        mod = OrderedModel(data['apply'].values.codes,
+                           np.asarray(data[columns], float),
+                           distr='logit')
+        cls.res = mod.fit(method='bfgs', disp=False)
+        # standard fit with pandas input
+        modp = OrderedModel(data['apply'],
+                            data[columns],
+                            distr='logit')
+        cls.resp = modp.fit(method='bfgs', disp=False)
+
+    def test_setup(self):
+        data = self.data
+        resp = self.resp
+        fittedvalues = resp.predict()
+
+        formulas = ["apply ~ 1 + pared + public + gpa + C(dummy)",
+                    "apply ~ pared + public + gpa + C(dummy)"]
+        for formula in formulas:
+            modf1 = OrderedModel.from_formula(formula, data, distr='logit')
+            resf1 = modf1.fit(method='bfgs')
+            summf1 = resf1.summary()
+            summf1_str = str(summf1)
+            assert resf1.model.exog_names == resp.model.exog_names
+            assert resf1.model.data.param_names == resp.model.exog_names
+            assert all(name in summf1_str for name in
+                       resp.model.data.param_names)
+            assert_allclose(resf1.predict(data[:5]), fittedvalues[:5])
+
+        # test over parameterized model with implicit constant
+        # warns but doesn't raise
+        formula = "apply ~ 0 + pared + public + gpa + C(dummy)"
+
+        with pytest.warns(SpecificationWarning):
+            modf2 = OrderedModel.from_formula(formula, data, distr='logit')
+
+        with pytest.warns(HessianInversionWarning):
+            resf2 = modf2.fit(method='bfgs')
+            assert np.isnan(resf2.bse).all()
```

Comment:
this assert fails on several, not all Windows environments.
It has finite values in those cases.

I have nans on my computer in this case, but I had other cases with finite values, although huge bse for unidentified parameters

Reply:
I'm not sure whether we get the HessianInversionWarning either, current assert fails before that
The optimization might stop at a parameter value that still has invertible Hessian

Reply:
```
        with pytest.warns(HessianInversionWarning):
>           resf2 = modf2.fit(method='bfgs')
E           Failed: DID NOT WARN. No warnings of type (<class 'statsmodels.tools.sm_exceptions.HessianInversionWarning'>,) was emitted. The list of emitted warnings is: [].
```